### PR TITLE
Fixed relational attribute import issue and other code

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Fixed relational attribute import issue and other code on the aci_destination_of_redirected_traffic resource.

### CODE REVIEW

File: aci/data_source_aci_vnsredirectdest.go [hunk 1]
```diff
@@ -80,5 +79,9 @@ func dataSourceAciDestinationofredirectedtrafficRead(ctx context.Context, d *sch
 	if err != nil {
 		return diag.FromErr(err)
 	}
+
+	// Importing vnsRsRedirectHealthGroup object
+	_, _ = getAndSetRemoteReadRelationvnsRsRedirectHealthGroup(aciClient, dn, d)
```
Comment (lhercot) [inline aci/data_source_aci_vnsredirectdest.go:84]: Do you need the left part can't you just call the method?

File: aci/resource_aci_vnsredirectdest.go [hunk 2]
```diff
@@ -84,7 +76,7 @@ func getRemoteDestinationofredirectedtraffic(client *client.Client, dn string) (
 	vnsRedirectDest := models.DestinationofredirectedtrafficFromContainer(vnsRedirectDestCont)
 
 	if vnsRedirectDest.DistinguishedName == "" {
-		return nil, fmt.Errorf("Destinationofredirectedtraffic %s not found", vnsRedirectDest.DistinguishedName)
+		return nil, fmt.Errorf("Destinationofredirectedtraffic %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:79]: I would add spaces in Destinationofredirectedtraffic

File: aci/resource_aci_vnsredirectdest.go [hunk 3]
```diff
@@ -130,13 +135,16 @@ func resourceAciDestinationofredirectedtrafficImport(d *schema.ResourceData, m i
...
-	pDN := GetParentDn(dn, fmt.Sprintf("/RedirectDest_ip-[%s]", ip))
+	pDN := GetParentDn(dn, fmt.Sprintf("/%s", fmt.Sprintf(models.RnvnsRedirectDest, ip)))
 	d.Set("service_redirect_policy_dn", pDN)
 	schemaFilled, err := setDestinationofredirectedtrafficAttributes(vnsRedirectDest, d)
 	if err != nil {
 		return nil, err
 	}
 
+	// Importing vnsRsRedirectHealthGroup object
+	_, _ = getAndSetRemoteReadRelationvnsRsRedirectHealthGroup(aciClient, dn, d)
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:146]: do you need the _, _ =?

File: aci/resource_aci_vnsredirectdest.go [hunk 4]
```diff
@@ -152,10 +160,13 @@ func resourceAciDestinationofredirectedtrafficCreate(ctx context.Context, d *sch
 	ServiceRedirectPolicyDn := d.Get("service_redirect_policy_dn").(string)
 
 	vnsRedirectDestAttr := models.DestinationofredirectedtrafficAttributes{}
+	annotation := ""
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:163]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 5]
```diff
@@ -152,10 +160,13 @@ func resourceAciDestinationofredirectedtrafficCreate(ctx context.Context, d *sch
...
 	if Annotation, ok := d.GetOk("annotation"); ok {
 		vnsRedirectDestAttr.Annotation = Annotation.(string)
+		annotation = Annotation.(string)
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:166]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 6]
```diff
@@ -152,10 +160,13 @@ func resourceAciDestinationofredirectedtrafficCreate(ctx context.Context, d *sch
...
 	} else {
 		vnsRedirectDestAttr.Annotation = "{}"
+		annotation = "{}"
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:169]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 7]
```diff
@@ -222,10 +234,13 @@ func resourceAciDestinationofredirectedtrafficUpdate(ctx context.Context, d *sch
 	ServiceRedirectPolicyDn := d.Get("service_redirect_policy_dn").(string)
 
 	vnsRedirectDestAttr := models.DestinationofredirectedtrafficAttributes{}
+	annotation := ""
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:237]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 8]
```diff
@@ -222,10 +234,13 @@ func resourceAciDestinationofredirectedtrafficUpdate(ctx context.Context, d *sch
...
 	if Annotation, ok := d.GetOk("annotation"); ok {
 		vnsRedirectDestAttr.Annotation = Annotation.(string)
+		annotation = Annotation.(string)
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:240]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 9]
```diff
@@ -222,10 +234,13 @@ func resourceAciDestinationofredirectedtrafficUpdate(ctx context.Context, d *sch
...
 	} else {
 		vnsRedirectDestAttr.Annotation = "{}"
+		annotation = "{}"
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:243]: Why do we need this?

File: aci/resource_aci_vnsredirectdest.go [hunk 10]
```diff
@@ -306,14 +322,9 @@ func resourceAciDestinationofredirectedtrafficRead(ctx context.Context, d *schem
...
-	vnsRsRedirectHealthGroupData, err := aciClient.ReadRelationvnsRsRedirectHealthGroupFromDestinationofredirectedtraffic(dn)
-	if err != nil {
-		log.Printf("[DEBUG] Error while reading relation vnsRsRedirectHealthGroup %v", err)
-		d.Set("relation_vns_rs_redirect_health_group", "")
 
-	} else {
-		setRelationAttribute(d, "relation_vns_rs_redirect_health_group", vnsRsRedirectHealthGroupData.(string))
-	}
+	// Importing vnsRsRedirectHealthGroup object
+	_, _ = getAndSetRemoteReadRelationvnsRsRedirectHealthGroup(aciClient, dn, d)
```
Comment (lhercot) [inline aci/resource_aci_vnsredirectdest.go:327]: do we need the "_, _ =" part?

File: examples/aci_destination_of_redirected_traffic/main.tf [hunk 11]
```diff
@@ -84,10 +84,10 @@ resource "aci_service_redirect_policy" "example" {
   relation_vns_rs_ipsla_monitoring_pol = aci_ip_sla_monitoring_policy.sla.id
 }
 
-resource "aci_destination_of_redirected_traffic" "example" {
-  service_redirect_policy_dn            = aci_service_redirect_policy.service_policy.id
-  ip                                    = "1.2.3.4"
-  ip2                                   = "10.20.30.40"
+resource "aci_destination_of_redirected_traffic" "l3_destinations_withou_mac" {
```
Comment (lhercot) [inline examples/aci_destination_of_redirected_traffic/main.tf:87]: without

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/data_source_aci_vnsredirectdest.go [hunk 12]
```diff
@@ -44,19 +45,16 @@ func dataSourceAciDestinationofredirectedtraffic() *schema.Resource {
...
+			"pod_id": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
 			},
-
-			"pod_id": &schema.Schema{
+			"relation_vns_rs_redirect_health_group": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
```
Comment (lhercot) [inline aci/data_source_aci_vnsredirectdest.go:55]: shouldn't this be computed?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/data_source_aci_vnsredirectdest.go (hunk 12 above)
Comment (sajagana) [reply aci/data_source_aci_vnsredirectdest.go:55]: It didn’t make difference in the state file.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**